### ISSUE TITLE
Multiple OpenCL GPUs not working together when farming

### ISSUE THREAD

**Issue by xieyusi9** [madMAx43v3r/chia-gigahorse#145]
Five AMD RX570 graphics cards are unable to work together. No matter how they are configured, only one card functions. It is unclear whether there is a misconfiguration or if there is a bug?

![image](https://github.com/madMAx43v3r/chia-gigahorse/assets/110649504/707e0f8f-f418-42d8-9a13-af746c6acb8d)
![image](https://github.com/madMAx43v3r/chia-gigahorse/assets/110649504/f19b0955-7ef1-42d9-a8f2-8a07c459ca75)

**Comment by xieyusi9** [comment]
I am using the complete package of version 1.8.1.giga13.

**Comment by Perfy** [comment]
Same problem with NVidia cards. Always work only card 1.
![2023-06-11_142711](https://github.com/madMAx43v3r/chia-gigahorse/assets/198541/f7124cc7-532d-4dda-8724-153d40106f40)

**Comment by ddubick** [comment]
*DISCLAIMER* Everything I'm going to say is guesswork as I don't have multiple CPUs.  I could definitely be wrong.

It says right at the top that it's using multiple GPU devices so what is making you think that it's only using 1 GPU?

If you guys are thinking used_gpu = 1 means that it's only using 1 GPU I think you're wrong.  I think that's a boolean saying that it used GPU to decompress as opposed to used_cpu = 1 and decompressing a single challenge wouldn't use more than 1 GPU anyway - pretty sure that multiple GPUs are for processing multiple challenges in parallel.  

Have you run GPU performance monitoring to prove that it's only using 1 GPU?

**Comment by xieyusi9** [comment]
@ddubick I have identified that only one graphics card is functioning through GPUZ.

**Comment by ddubick** [comment]
So another guess here... could it be that you've got CHIAPOS_MAX_CORES set to 2?  The definition of this variable states that it controls the number of parallel processes.

> When using GPU(s) to farm, it is recommended to allocate 2 cores per low end GPU, 4 cores per mid range GPU and 8 cores per high end GPU. With faster CPUs you can use less cores, and thus less RAM.

**Comment by Perfy** [comment]
I built a recompute server with a quad-core CPU and two GPUs.
Both GPUs show the load, but on the first one it is much more.
The log is still "used_gpu = 1", which really means the number of GPUs used in the calculation.

It would be more useful to see the index of the used GPU.

**Comment by madMAx43v3r** [comment]
> So another guess here... could it be that you've got CHIAPOS_MAX_CORES set to 2? The definition of this variable states that it controls the number of parallel processes.

Yes CHIAPOS_MAX_CORES needs to be at least as large as the number of GPUs, with big GPUs it should be 2x or 4x more.

>  The log is still "used_gpu = 1", which really means the number of GPUs used in the calculation.

This is just a boolean, 0 = no GPU used, 1 = one GPU was used

**Comment by xieyusi9** [comment]
@madMAx43v3r These graphics card mining rigs were previously used for Ethereum mining and have since been phased out. The CPU of the rig only has two cores.

**Comment by madMAx43v3r** [comment]
Yes that's fine, you can still use more "cores". It may not be enough for a lot of GPUs though...

**Comment by nqzhang** [comment]
i have same question,Is there any way to use all gpu? Even if the cpu performance is poor

Jun 24 23:20:36 hive177_30802n chia_recompute_server[26329]: [chiapos] Using 2 / 2 CPU threads
Jun 24 23:20:36 hive177_30802n chia_recompute_server[26329]: [chiapos] Using 2 / 2 CUDA devices
Jun 24 23:20:37 hive177_30802n chia_recompute_server[26329]: [chiapos] Using 0 / 0 OpenCL devices (AMD Accelerated Parallel Processing)
Jun 24 23:20:37 hive177_30802n chia_recompute_server[26329]: Using 2 / 2 CPU threads
Jun 24 23:20:37 hive177_30802n chia_recompute_server[26329]: Using NVIDIA GeForce RTX 3080 (CUDA (native))
Jun 24 23:20:37 hive177_30802n chia_recompute_server[26329]: ReComputeServer listening on 0.0.0.0 port 11989
Jun 24 23:20:45 hive177_30802n chia_recompute_server[26329]: [1687620045395] Request from 192.168.2.111 for K32 C8 took 1008.79 ms (used_gpu = 1, is_fail = 0)
Jun 24 23:20:46 hive177_30802n chia_recompute_server[26329]: [1687620046022] Request from 192.168.2.111 for K32 C8 took 1595.18 ms (used_gpu = 1, is_fail = 0)
Jun 24 23:20:46 hive177_30802n chia_recompute_server[26329]: [1687620046612] Request from 192.168.2.203 for K32 C8 took 998.49 ms (used_gpu = 1, is_fail = 0)

**Comment by madMAx43v3r** [comment]
Yes, increase `CHIAPOS_MAX_CORES` to 4 or 8
